### ISSUE TITLE
empty-table-header virtualised rule should run without requiring role

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3788]
This test needs to be added, and needs to pass.

```js
  it('should fail if table header has no child nodes', function () {
    var node = new axe.SerialVirtualNode({
      nodeName: 'th',
      attributes: {}
    });
    node.children = [];

    var results = axe.runVirtualRule('empty-table-header', node);

    assert.lengthOf(results.passes, 0);
    assert.lengthOf(results.violations, 1);
    assert.lengthOf(results.incomplete, 0);
  });
```

**Comment by dbowling** [comment]
@WilcoFiers - as I was looking at this tonight, I see we actually [already have this test as part of my original PR](https://github.com/dequelabs/axe-core/pull/3621/files?diff=split&w=1#diff-8b7797425f99ada52d4cfcff689f3e8b56e74b5c84e90eba0c440774b33a4d3eR2-R13), but our expectation was that it incompletes:

```JavaScript
it('should incomplete when children are missing', function () {
    var thNode = new axe.SerialVirtualNode({
      nodeName: 'th'
    });
    thNode.children = [];

    var results = axe.runVirtualRule('empty-table-header', thNode);

    assert.lengthOf(results.passes, 0);
    assert.lengthOf(results.violations, 0);
    assert.lengthOf(results.incomplete, 1);
  });
```

**Note**: existing test is has `attributes: {}` vs `attributes: undefined`, and has different asserts for `results.violations` and `results.incomplete`.

The test was based on [a suggestion](https://github.com/dequelabs/axe-core/pull/3621/files?diff=split&w=1#r956142997) from Steve:

> We'll want a test that should return incomplete if the `th` element does not have the children set to an empty array. Similar to [button-name](https://github.com/dequelabs/axe-core/blob/develop/test/integration/virtual-rules/button-name.js#L116-L125) 

TTBOMK, the empty object vs `undefined` for `attributes` isn't something that this code should differentiate on, so I'm not seeing how things need to change. Let me know if that's not the case and I will look with fresh eyes, and maybe we can chat about how to make a test that is more expressive about intent.

**Comment by straker** [comment]
We dug into this. The reason this test is not failing is due to the [role algorithm](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/aria/get-role.js#L56-L60) looking up the parent tree of the `th` looking for presentational parents. That is because if the table parent was presentational, the entire table is and we don't want to run the `empty-table-header` rule on it. 

The fix is to not throw on that line if we don't have an `actualNode`, which will then allow the test to fail as expected. 

Looking at this we also discovered that the `empty-table-header` rule is looking at `th` elements even if their role is changed, so we should update the selector to exclude `th` elements with a role: `th:not([role])`

**Comment by padmavemulapati** [comment]
Validated with the latest code base, axe-core develop branch:
for the test snippet:
```
<table>
    <tr>
      <th id="ignore1" role="spinbutton">rowheader with a role</th>
      <th id="ignore1" role="none">rowheader with a role</th>
      <th id="ignore1" role="cell">rowheader with a role</th>
    </tr>
  </table>
```
`empty-table-header` `fails` when `role="spinbutton"` , `pass` when `role="none" or role="cell"`



![Image](https://user-images.githubusercontent.com/47409579/206438048-1a378be1-c3f8-4425-98e6-fb935c50a8fa.png)
